File: frontend/services/postService.js
import localStoragePostRepository from "../repositories/posts/localStoragePostRepository";
import remotePostRepository from "../repositories/posts/remotePostRepository";
import forceDelay from "../utils/forceDelay";
import {error} from "next/dist/build/output/log";


const getAllPosts = async ({token, board, page, pageSize ,search}) => {

    if (!token) {
        return await localStoragePostRepository.getAllPosts({
            page, pageSize, board , search
        });
    }

    return await remotePostRepository.getAllPosts({token, page, pageSize, board , search});
}

const createPost = async ({token, post}) => {

    if (!token) {
        return await localStoragePostRepository.createPost(post);
    }

    return await remotePostRepository.createPost({token, post});
}


const removePost = async ({token, id}) => {

    if (!token) {
        return await localStoragePostRepository.removePost(id);
    }

    return await remotePostRepository.removePost({token, id});


}

const postService = {
    getAllPosts,
    createPost,
    removePost
}

export default postService;

File: backend/src/main/java/com/pjatk/tin/postitboard/backend/service/PostService.java
package com.pjatk.tin.postitboard.backend.service;

import com.pjatk.tin.postitboard.backend.controller.request.AddPostToBoardRequest;
import com.pjatk.tin.postitboard.backend.controller.response.PostsByBoardResponse;
import com.pjatk.tin.postitboard.backend.model.Board;
import com.pjatk.tin.postitboard.backend.model.Post;
import com.pjatk.tin.postitboard.backend.model.User;
import com.pjatk.tin.postitboard.backend.repository.PostRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;


@Service
@RequiredArgsConstructor
@Slf4j
public class PostService {

    private final PostRepository postRepository;
    private final BoardService boardService;
    private final SecurityService securityService;

    public PostsByBoardResponse getPosts(Long boardId, int pageSize, int page, String search) {

        Page<Post> posts = postRepository.findAllByBoardIdAndContentContains(boardId, search, PageRequest.of(page, pageSize));

        return PostsByBoardResponse.from(posts);
    }


    @Transactional
    public void createPost(AddPostToBoardRequest request) {

        Board board = boardService.getBoard(request.getBoard());

        User creator = securityService.getCurrentUser();

        Post post = Post.builder()
                .board(board)
                .creator(creator)
                .content(request.getContent())
                .color(request.getColor())
                .build();

        postRepository.save(post);

    }

    @Transactional
    public void deletePost(Long postId) {
        log.info("Deleting post with id: {}", postId);
        postRepository.deleteById(postId);
    }


}
